staging/src/k8s.io/csi-translation-lib/plugins/azure_file.go

```diff
@@ -163,25 +162,37 @@ func (t *azureFileCSITranslator) TranslateCSIPVToInTree(pv *v1.PersistentVolume)
 		ReadOnly: csiSource.ReadOnly,
 	}
 
+	for k, v := range csiSource.VolumeAttributes {
+		switch strings.ToLower(k) {
+		case shareNameField:
+			azureSource.ShareName = v
+		case secretNamespaceField:
```

Comment:
When does the CSI driver set this field? Can we have it use only NodeStageSecretRef?

Reply:
It's up to user, user could use `NodeStageSecretRef` or use `secretName` parameter to set secret, anyway, we supports both.

Reply:
Can we deprecate the use of `secretName` in the CSI driver since it is equivalent to `NodeStageSecretRef`? The advantage of NodeStageSecretRef is that the driver does not need broad permissions for accessing Kubernetes secrets.

Reply:
Yes, in CSI driver, we could suggest using `NodeStageSecretRef`, while in this translation lib, we still need to handle such scenario.